internal/api/api.go

```diff
@@ -36,15 +38,33 @@ func NewHTTPClient() CertAPIClient {
 	return CertAPIClient{Client: &http.Client{}}
 }
 
+type catalogQueryResponse struct {
+	Page     uint          `json:"page"`
+	PageSize uint          `json:"page_size"`
+	Total    uint          `json:"total"`
+	Data     []interface{} `json:"data"`
+}
+
 // IsContainerCertified get container image info by repo/name and checks if container details is present
 // If present then returns `true` as certified operators.
-func (api CertAPIClient) IsContainerCertified(repository, imageName string) (bool, error) {
-	imageID, err := api.GetImageIDByRepository(repository, imageName)
+func (api CertAPIClient) IsContainerCertified(id configsections.ContainerImageIdentifier) (bool, error) {
+	var url string
+	const defaultTag = "latest"
+	if id.Digest == "" {
+		if id.Tag == "" {
+			id.Tag = defaultTag
+		}
+		url = fmt.Sprintf("%s/%s/%s/images?page_size=1&filter=repositories.repository==%s/%s;repositories.tags.name==%s", apiCatalogByRepositoriesBaseEndPoint, id.Repository, id.Name, id.Repository, id.Name, id.Tag)
+	} else {
+		url = fmt.Sprintf("%s/%s/%s/images?page_size=1&filter=image_id==%s", apiCatalogByRepositoriesBaseEndPoint, id.Repository, id.Name, id.Digest)
```

Comment:
Maybe this section should be made into its own function to properly build the URL.  Easy to unit test as well.

Reply:
Done